tests/conftest.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2015, 2016 CERN.
+#
+# Invenio is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+
+"""Pytest configuration."""
+
+from __future__ import absolute_import, print_function
+
+import os
+import shutil
+import tempfile
+
+import pytest
+from flask import Flask
+from invenio_db import InvenioDB, db
+from sqlalchemy_utils.functions import create_database, database_exists, \
+    drop_database
+
+
+@pytest.fixture
+def app(request):
+    """Flask application fixture."""
+    instance_path = tempfile.mkdtemp()
+    app = Flask('testapp')
+    app.config.update(
+        SQLALCHEMY_DATABASE_URI=os.getenv('SQLALCHEMY_DATABASE_URI',
+                                          'sqlite://'),
+    )
+    InvenioDB(app)
+
+    with app.app_context():
+        if str(db.engine.url) != 'sqlite://' and \
+                not database_exists(str(db.engine.url)):
+            create_database(str(db.engine.url))
+        db.create_all()
+
+    def teardown():
+        with app.app_context():
+            db.session.close()
+            if str(db.engine.url) != 'sqlite://':
+                drop_database(str(db.engine.url))
+            shutil.rmtree(instance_path)
+
+    request.addfinalizer(teardown)
+
+    app.test_request_context().push()
```

Comment:
please have a look at `@pytest.yield_fixture` and example in https://github.com/inveniosoftware/invenio-records/blob/master/tests/conftest.py#L45-L73